### PR TITLE
Update dg design

### LINKED ISSUE #150: [PE-D] Missing section in User Guide
The User Guide's section 2.2 is incomplete. There could have been at least some content since the team has already managed to put together some sort of a GUI?
Screenshot 2020-10-30 at 4.17.58 PM

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: foojingyi/ped#2

### PR DESCRIPTION
**Pull request by AlexCQY**
-update Model and Ui component diagrams
-re-insert schedule tab into GUI (It was accidentally taken out)
-include jfxtras lib acknowledgements in readme
-update AddCommand in DG

### CODE REVIEW

File: docs/diagrams/AcademicClassDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+Class Academic
+Class Question
+Class Exam
+Class Attendance
+}
+
+
+Academic *--> "*" Question
+Academic *--> "*" Exam
+Academic *--> "*" Attendance
```
Comment (ong-yinggao98) [inline docs/diagrams/AcademicClassDiagram.puml:19]: Academic is actually more of an aggregation than a composition, Questions, Exams and Attendance are all created before the Academic class is created, and they can exist without it. I think you could just stick to using a normal association.

File: docs/diagrams/AdminClassDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+Class Fee
+Class PaymentDate
+Class Details
+}
+
+Admin *--> ClassVenue
+Admin *--> ClassTime
+Admin *--> Fee
+Admin *--> PaymentDate
+Admin *--> "*" Details
```
Comment (ong-yinggao98) [inline docs/diagrams/AdminClassDiagram.puml:20]: Same for Admin

File: docs/diagrams/ModelClassDiagram.puml [hunk 3]
```diff
@@ -5,52 +5,58 @@ skinparam arrowColor MODEL_COLOR
...
 }
 
-Package Tag {
-Class Tag
+Package Admin {
+Class Admin
+}
+
+Package Academic {
+Class Academic
```
Comment (ong-yinggao98) [inline docs/diagrams/ModelClassDiagram.puml]: Admin and Academic are actually packages within Student, could you perhaps nest them within the Student package?

**Review by ong-yinggao98 — CHANGES_REQUESTED** [review]
I've left some comments, do check em out

File: docs/diagrams/ModelClassDiagram.puml (hunk 3 above)
Comment (AlexCQY) [reply docs/diagrams/ModelClassDiagram.puml]: oh yea thanks

File: docs/diagrams/AcademicClassDiagram.puml (hunk 1 above)
Comment (AlexCQY) [reply docs/diagrams/AcademicClassDiagram.puml:19]: hmmm.. then should `Student` and its field's relationship also be association?

File: docs/diagrams/AcademicClassDiagram.puml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+@startuml
+!include style.puml
+skinparam arrowThickness 1.1
+skinparam arrowColor MODEL_COLOR
+skinparam classBackgroundColor MODEL_COLOR
+
+Package Academic {
+Class Academic
+Class Question
```
Comment (ong-yinggao98) [inline docs/diagrams/AcademicClassDiagram.puml]: Is there a way to show that Question is an abstract class with UnsolvedQuestion and SolvedQuestion inheriting from it?

**Review by ong-yinggao98 — APPROVED** [review]
Overall looks good, just look at the comment if interested and merge it when you are ready.

File: docs/diagrams/AcademicClassDiagram.puml (hunk 4 above)
Comment (AlexCQY) [reply docs/diagrams/AcademicClassDiagram.puml]: updated it 
image

**Pull request merged**